app/services/paypal_service/api/accounts.rb

```diff
@@ -68,17 +63,45 @@ def create(community_id:, person_id: nil, order_permission_request_token:, body:
               token_secret: access_token[:token_secret]
             })
         ) { |personal_data|
-          account = PaypalAccountStore.update(
+          opts = {
+            email: personal_data[:email],
+            payer_id: personal_data[:payer_id],
+            order_permission_verification_code: body[:order_permission_verification_code],
+            order_permission_scope: access_token[:scope].join(','),
+            active: true
+          }
+
+          existing = PaypalAccountStore.get(
             community_id: community_id,
             person_id: person_id,
-            opts:
-              {
-                email: personal_data[:email],
+            payer_id: personal_data[:payer_id]
+          )
+
+          account =
+            if existing.nil?
+              # Update the 'new' account
+              PaypalAccountStore.update(
+                community_id: community_id,
+                person_id: person_id,
+                order_permission_request_token: order_permission_request_token,
+                opts: opts
+              )
+            else
+              # Update the 'existing' account
+              PaypalAccountStore.update(
+                community_id: community_id,
+                person_id: person_id,
                 payer_id: personal_data[:payer_id],
-                order_permission_verification_code: body[:order_permission_verification_code],
-                order_permission_scope: access_token[:scope].join(','),
-                active: true
-              })
+                opts: opts
+              )
+
+              # Delete the 'new' account
+              PaypalAccountStore.delete(
+                community_id: community_id,
+                person_id: person_id,
+                order_permission_request_token: order_permission_request_token
+              )
+            end
```

Comment:
This is a bit too long to be readable. Pull the if-part to a private helper, something like: `create_verified_account(opts, existing)`. That way you can also eliminate the opts variable and declare the data directly as call parameter. Also, there might be a bug in the else branch because it returns the result of delete operation, not create operation.